### ISSUE TITLE
webpack4 Entrypoint undefined = index.html

### ISSUE THREAD

**Issue by StringKe** [SimulatedGREG/electron-vue#828]
### logs
```
/Users/chen/.nvm/versions/node/v10.15.2/bin/node /Users/chen/.nvm/versions/node/v10.15.2/lib/node_modules/npm/bin/npm-cli.js run build --scripts-prepend-node-path=auto

> browser@0.0.1 build /Users/chen/Desktop/vueBrowser
> node .electron-vue/build.js && electron-builder


  lets-build

  - building main process
  - building renderer process
(node:91724) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead
  ✔ building main process
  \ building renderer process

   ERROR  failed to build renderer process

    Hash: 94e8f68bfdc8762843fd
    Version: webpack 4.29.6
    Time: 7462ms
    Built at: 03/20/2019 3:52:11 PM
                         Asset     Size  Chunks  Chunk Names
     fonts/ionicons--fonts.ttf  193 KiB          
    fonts/ionicons--fonts.woff  193 KiB          
      imgs/ionicons--fonts.svg  542 KiB          
                   renderer.js  577 KiB       0  renderer
    Entrypoint renderer = renderer.js
      [0] ./node_modules/vue-loader/lib/runtime/componentNormalizer.js 2.63 KiB {0} [built]
      [4] ./node_modules/vue-style-loader/lib/addStylesClient.js + 1 modules 6.71 KiB {0} [built]
          | ./node_modules/vue-style-loader/lib/addStylesClient.js 6.09 KiB [built]
          | ./node_modules/vue-style-loader/lib/listToStyles.js 637 bytes [built]
     [10] ./src/constants.js 452 bytes {0} [built]
     [18] ./src/renderer/components/pivatePage/UserCenterPage.vue + 4 modules 1.43 KiB {0} [built]
          | ./src/renderer/components/pivatePage/UserCenterPage.vue 485 bytes [built]
          | ./src/renderer/components/pivatePage/UserCenterPage.vue?vue&type=template&id=f0ef0fc2& 224 bytes [built]
          | ./src/renderer/components/pivatePage/UserCenterPage.vue?vue&type=script&lang=js& 390 bytes [built]
          | ./node_modules/vue-loader/lib/loaders/templateLoader.js??vue-loader-options!./node_modules/vue-loader/lib??vue-loader-options!./src/renderer/components/pivatePage/UserCenterPage.vue?vue&type=template&id=f0ef0fc2& 200 bytes [built]
          | ./node_modules/babel-loader/lib!./node_modules/vue-loader/lib??vue-loader-options!./src/renderer/components/pivatePage/UserCenterPage.vue?vue&type=script&lang=js& 162 bytes [built]
     [25] external "dayjs" 42 bytes {0} [built]
     [38] external "path" 42 bytes {0} [built]
     [49] ./src/renderer/store/modules/index.js 238 bytes {0} [built]
     [50] ./src/renderer/components/BrowserPage/Browser/tabs/pane.vue + 4 modules 2.43 KiB {0} [built]
          | ./src/renderer/components/BrowserPage/Browser/tabs/pane.vue 461 bytes [built]
          | ./src/renderer/components/BrowserPage/Browser/tabs/pane.vue?vue&type=template&id=0a3d0268& 226 bytes [built]
          | ./src/renderer/components/BrowserPage/Browser/tabs/pane.vue?vue&type=script&lang=js& 394 bytes [built]
          | ./node_modules/vue-loader/lib/loaders/templateLoader.js??vue-loader-options!./node_modules/vue-loader/lib??vue-loader-options!./src/renderer/components/BrowserPage/Browser/tabs/pane.vue?vue&type=template&id=0a3d0268& 243 bytes [built]
          | ./node_modules/babel-loader/lib!./node_modules/vue-loader/lib??vue-loader-options!./src/renderer/components/BrowserPage/Browser/tabs/pane.vue?vue&type=script&lang=js& 1.13 KiB [built]
     [89] external "axios" 42 bytes {0} [built]
     [90] external "iview" 42 bytes {0} [built]
     [91] ./node_modules/dayjs/locale/es.js 1.04 KiB {0} [built]
    [100] ./node_modules/iview/dist/styles/iview.css 433 bytes {0} [built]
    [106] ./src/renderer/assets/css/style.css 465 bytes {0} [built]
    [202] external "vue-electron" 42 bytes {0} [built]
    [203] ./src/renderer/main.js + 66 modules 442 KiB {0} [built]
          | ./src/renderer/main.js 1.49 KiB [built]
          | ./node_modules/vue/dist/vue.esm.js 318 KiB [built]
          | ./src/renderer/App.vue 446 bytes [built]
          | ./src/renderer/router/index.js 1.95 KiB [built]
          | ./src/renderer/store/index.js 335 bytes [built]
          | ./src/datastore.js 2.74 KiB [built]
          | ./src/renderer/components/BrowserPage/Browser/tabs/index.js 101 bytes [built]
          | ./src/renderer/components/BrowserPage/Browser/AutoComplete.vue 555 bytes [built]
          | ./src/renderer/App.vue?vue&type=template&id=b9fc48f0& 201 bytes [built]
          | ./src/renderer/App.vue?vue&type=script&lang=js& 344 bytes [built]
          | ./src/renderer/components/BrowserPage/Browser/tabs/tabs.vue 461 bytes [built]
          | ./src/renderer/components/BrowserPage/Browser/AutoComplete.vue?vue&type=template&id=1e14fd3c& 228 bytes [built]
          | ./src/renderer/components/BrowserPage/Browser/AutoComplete.vue?vue&type=script&lang=js& 398 bytes [built]
          | ./node_modules/vue-loader/lib/loaders/templateLoader.js??vue-loader-options!./node_modules/vue-loader/lib??vue-loader-options!./src/renderer/App.vue?vue&type=template&id=b9fc48f0& 210 bytes [built]
          | ./node_modules/babel-loader/lib!./node_modules/vue-loader/lib??vue-loader-options!./src/renderer/App.vue?vue&type=script&lang=js& 73 bytes [built]
          |     + 52 hidden modules
        + 195 hidden modules
    
 >  ERROR in unknown: Unexpected token (22388:8)
 >  Child html-webpack-plugin for "index.html":
 >          Asset     Size  Chunks  Chunk Names
 >       index.html  556 KiB       0  
 >       Entrypoint undefined = index.html
        [0] ./node_modules/html-webpack-plugin/lib/loader.js!./src/index.ejs 1.21 KiB {0} [built]
        [1] ./node_modules/lodash/lodash.js 527 KiB {0} [built]
        [2] (webpack)/buildin/module.js 497 bytes {0} [built]


npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! browser@0.0.1 build: `node .electron-vue/build.js && electron-builder`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the browser@0.0.1 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/chen/.npm/_logs/2019-03-20T07_52_11_348Z-debug.log

Process finished with exit code 1
```
I found ["Entrypoint undefined"](https://github.com/jantimon/html-webpack-plugin/issues/850) but I still don't know how to fix this error.

**Comment by miya-yang** [comment]
me too

**Comment by RockyYui** [comment]
兄弟，请问你怎么解决的啊？跟你一模一样的错误，求教

**Comment by StringKe** [comment]
> 兄弟，请问你怎么解决的啊？跟你一模一样的错误，求教

没有解决

**Comment by StringKe** [comment]
> 兄弟，请问你怎么解决的啊？跟你一模一样的错误，求教

没有解决

**Comment by DREZZ3R** [comment]
I had a similar problem.  In development `yarn dev` was working, but broke the build. My problem was how I imported the package. I solved with this:
````javascript
import Excel from "exceljs/modern.browser"; // Breaks Build
import Excel from "exceljs"; // Didn't break the build
````

My debug process was:
1 - Create a clean electron-vue project
2 - Copy ONE page/view (the components that are in the `/src/routes/index.js`) from the project with broken build to the new project.
3 - Build it
4 - Repeat process step 2 and 3 until it the build fails
5 - Once I know which component is giving trouble, remove all content from the `<template></temlate>` and just leave the `imports` like this:

````
<template></template>
<script>
import package1 from 'package1'
import package2 from 'package2'
import package3 from 'package2'
export default {
}
</script>
````

6 - remove/comment one import and build it. Comment another one and built it. Do this process until the build is successfull.

7 - Once you know which package is giving trouble try to check what is causing the problem. In my case was the previous mentioned `exceljs` import

**TL:DR**
- Create a clean project
- Copy the smallest quantity of files from your older project to the new one and build.
- Keep adding other files until the project build breaks;
- In one of those files/file there should be the problem.

**Comment by yh4922** [comment]
https://github.com/SimulatedGREG/electron-vue/issues/1022#issuecomment-644701614
